src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+
+    Access::SubjectDescriptor descriptor = ctx.mCommandHandler.GetSubjectDescriptor();
+    if (AuthMode::kCase == descriptor.authMode)
+    {
+        ReturnOnFailure(AddResponseOnError<ResponseType>(
+            ctx, response, Attributes::LastConfiguredBy::Set(ctx.mRequestPath.mEndpointId, descriptor.subject)));
+    }
+    else
+    {
+        ReturnOnFailure(
+            AddResponseOnError<ResponseType>(ctx, response, Attributes::LastConfiguredBy::SetNull(ctx.mRequestPath.mEndpointId)));
+    }
+
+    // Write response
+    response.status = to_underlying(Protocols::InteractionModel::Status::Success);
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+template <typename CommandData, typename ResponseType>
+void ViewSceneParse(HandlerContext & ctx, const CommandData & req, GroupDataProvider * groupProvider)
+{
+    ResponseType response;
+
+    uint16_t endpoinTableSize = 0;
+
+    ReturnOnFailure(AddResponseOnError<ResponseType>(
+        ctx, response, Attributes::SceneTableSize::Get(ctx.mRequestPath.mEndpointId, &endpoinTableSize)));
+
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable =
+        chip::scenes::GetSceneTableImpl(ctx.mRequestPath.mEndpointId, endpoinTableSize);
+
+    // Response data
+    response.groupID = req.groupID;
+    response.sceneID = req.sceneID;
+
+    // Verify Endpoint in group
+    VerifyOrReturn(nullptr != groupProvider);
+    if (0 != req.groupID &&
+        !groupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupID, ctx.mRequestPath.mEndpointId))
+    {
+        response.status = to_underlying(Protocols::InteractionModel::Status::InvalidCommand);
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        return;
+    }
+
+    SceneTableEntry scene;
+
+    //  Gets the scene form the table
+    ReturnOnFailure(
+        AddResponseOnError<ResponseType>(ctx, response,
+                                         sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(),
+                                                                        SceneStorageId(req.sceneID, req.groupID), scene)));
+
+    // Verifies how to convert transition time
+    if (scene.mStorageData.mSceneTransitionTimeMs)
+    {
+        if (Commands::ViewScene::Id == ctx.mRequestPath.mCommandId)
+        {
+            response.transitionTime.SetValue(static_cast<uint16_t>(scene.mStorageData.mSceneTransitionTimeMs / 1000));
+        }
+        else if (Commands::EnhancedViewScene::Id == ctx.mRequestPath.mCommandId)
+        {
+            response.transitionTime.SetValue(static_cast<uint16_t>(scene.mStorageData.mSceneTransitionTimeMs / 100));
+        }
+    }
+
+    // Verify if name was saved in the scene
+    if (scene.mStorageData.mNameLength)
+    {
+        response.sceneName.SetValue(CharSpan(scene.mStorageData.mName, scenes::kSceneNameMaxLength));
```

Comment:
This is going to send garbage data on the wire, no?  Why aren't we setting the _actual correct length_ that we have right above?